src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -386,6 +386,56 @@ protected override void MethodChecks(BindingDiagnosticBag diagnostics)
 
         protected sealed override void ExtensionMethodChecks(BindingDiagnosticBag diagnostics)
         {
+            if (ContainingType is { IsExtension: true, ExtensionParameter: { Type.IsStatic: false } extensionParameter } && !IsStatic &&
+                OperatorFacts.IsCompoundAssignmentOperatorName(Name))
+            {
+                if (extensionParameter.Name == "")
+                {
+                    diagnostics.Add(ErrorCode.ERR_InstanceMemberWithUnnamedExtensionsParameter, _location, Name);
+                }
+
+                // PROTOTYPE: confirm the following rule
+
+                // Require receiver type to be known as a class or as a struct, and
+                // require:
+                //     - struct receiver to be a 'ref' 
+                //     - class receiver to be 'by val' (not 'ref', not 'in', not 'ref readonly')
+
+                // We don't report every invalid combination here in order to avoid producing too much noise.
+                switch (extensionParameter.RefKind)
+                {
+                    case RefKind.Out: // 'out' is disallowed in general
+                    case RefKind.Ref: // 'ref' receivers are disallowed for types not known to be a struct
+                        break;
+
+                    case RefKind.In:
+                    case RefKind.RefReadOnlyParameter:
+                        // 'in' and 'ref readonly' receivers are disallowed for anything that is not a concrete struct (class or a type parameter)
+                        if (extensionParameter.Type is { TypeKind: not TypeKind.TypeParameter, IsValueType: true })
```

Comment:
>extensionParameter.Type is { TypeKind: not TypeKind.TypeParameter, IsValueType: true } [](http://example.com/codeflow?start=28&length=86)

Would it be simpler to check `.TypeKind == TypeKind.Struct`?

Reply:
Perhaps